updater/dart_cli/pubspec.lock

```diff
@@ -0,0 +1,381 @@
+# Generated by pub
```

Comment:
Thoughts on adding the `pubspec.lock` to the `.gitignore` since I believe the lock file is not taken into consideration when the CLI is activated via pub.

Reply:
I suspect it doesn't matter here much.  Technically this should commit today, but won't once we move it to be a library? https://dart.dev/guides/libraries/private-files

Reply:
I think Dart CLIs are treated like libraries.

> For library packages, don’t commit the pubspec.lock file. Regenerating the pubspec.lock file lets you test your package against the latest compatible versions of its dependencies.

Other CLIs like `pub` don't commit the lock file either https://github.com/dart-lang/pub but yeah it's not a big deal either way.